enterprise/kernel/src/main/java/org/neo4j/kernel/impl/enterprise/SecurityLog.java

```diff
@@ -38,35 +39,31 @@
 import static org.neo4j.helpers.Strings.escape;
 import static org.neo4j.kernel.impl.enterprise.configuration.EnterpriseEditionSettings.security_log_filename;
 
-public class SecurityLog implements Log
+public class SecurityLog implements Log, Lifecycle
 {
+    private RotatingFileOutputStreamSupplier rotatingSupplier;
     private final Log inner;
 
     public SecurityLog( Config config, FileSystemAbstraction fileSystem, Executor executor ) throws IOException
     {
-        this( createLog( config, fileSystem, executor ) );
-    }
-
-    public SecurityLog( Log log )
-    {
-        inner = log;
-    }
```

Comment:
I reviewed the usage and the only non-test place it is used it gets from EditionModel.authManagerLog, which is always either the real security log, or a null log. So this means we're back to two cases, a closable log or a non-closable one, and that is dealt with in the condition inside the shutdown method.

I would prefer a better approach, but the use of two constructors for setting up SecurityLog is a pre-existing design issue I did not want to resolve in this PR, which focuses only on adding the close() call to the real security log (and not to test or null-logs).